tools/building.py

```diff
@@ -1565,6 +1570,8 @@ def run_binaryen_command(tool, infile, outfile=None, args=[], debug=False, stdou
     #       support them by emitting a source map at the end from the dwarf,
     #       and use llvm-objcopy to remove that final dwarf
     cmd += ['--strip-dwarf']
+  if outfile and tool == 'wasm-opt' and Settings.MEMORY64 == 1:
+    cmd += ['--memory64-lowering']
```

Comment:
I don't think this is the right place. This should be in emcc.py alongside the other pass decisions we make for the main wasm-opt call (see where we add `--safe-heap`). By putting it here, this pass will be run in every wasm-opt invocation (like for import/export minification), most of them unnecessarily.